src/controller/python/chip/clusters/attribute.cpp

```diff
@@ -225,23 +225,25 @@ class ReadClientCallback : public ReadClient::Callback
 
     void OnReportEnd() override { gOnReportEndCallback(mAppContext); }
 
-    void OnDone(ReadClient *) override
+    void OnDone(ReadClient * apReadClient) override
     {
         gOnReadDoneCallback(mAppContext);
 
-        delete this;
+        // SubscriptionType will delete with pychip_ReadClient_Abort
+        if (apReadClient->IsReadType())
+        {
+            delete apReadClient;
```

Comment:
the change of not having apReadClient as a member of the class anymore is not described in the summary of the PR.

Please explain changes and why in the summary of the PR.

Reply:
Has been restored here